axelor-sale/src/main/java/com/axelor/apps/sale/service/saleorder/SaleOrderInitValueServiceImpl.java

```diff
@@ -0,0 +1,152 @@
+package com.axelor.apps.sale.service.saleorder;
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.base.db.Company;
+import com.axelor.apps.base.db.Partner;
+import com.axelor.apps.base.db.PrintingSettings;
+import com.axelor.apps.base.db.repo.CompanyRepository;
+import com.axelor.apps.base.service.BankDetailsService;
+import com.axelor.apps.base.service.app.AppBaseService;
+import com.axelor.apps.base.service.user.UserService;
+import com.axelor.apps.sale.db.SaleOrder;
+import com.axelor.apps.sale.db.repo.SaleOrderRepository;
+import com.axelor.apps.sale.service.app.AppSaleService;
+import com.axelor.auth.AuthUtils;
+import com.axelor.auth.db.User;
+import com.axelor.db.mapper.Mapper;
+import com.axelor.studio.db.AppBase;
+import com.axelor.studio.db.AppSale;
+import com.google.inject.Inject;
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class SaleOrderInitValueServiceImpl implements SaleOrderInitValueService {
+
+  protected AppBaseService appBaseService;
+  protected CompanyRepository companyRepository;
+  protected AppSaleService appSaleService;
+  protected UserService userService;
+  protected BankDetailsService bankDetailsService;
+
+  @Inject
+  public SaleOrderInitValueServiceImpl(
+      AppBaseService appBaseService,
+      CompanyRepository companyRepository,
+      AppSaleService appSaleService,
+      UserService userService,
+      BankDetailsService bankDetailsService) {
+    this.appBaseService = appBaseService;
+    this.companyRepository = companyRepository;
+    this.appSaleService = appSaleService;
+    this.userService = userService;
+    this.bankDetailsService = bankDetailsService;
+  }
+
+  @Override
+  public Map<String, Object> getOnNewInitValues(SaleOrder saleOrder) throws AxelorException {
+    Map<String, Object> initValues = new HashMap<>();
+    initValues.putAll(saleOrderDefaultValues(saleOrder));
+    initValues.putAll(getInAti(saleOrder));
+    initValues.putAll(getBankDetails(saleOrder));
+    initValues.putAll(getGroupProductsOnPrintings(saleOrder));
+    return initValues;
+  }
+
+  protected Map<String, Object> saleOrderDefaultValues(SaleOrder saleOrder) {
+    Map<String, Object> saleOrderMap = new HashMap<>();
+    AppSale appSale = appSaleService.getAppSale();
+    Company company = getCompany();
+    int salesPersonSelect = appSale.getSalespersonSelect();
+    User user = AuthUtils.getUser();
+    PrintingSettings printingSettings =
+        Optional.ofNullable(user)
+            .map(User::getActiveCompany)
+            .map(Company::getPrintingSettings)
+            .orElse(null);
+
+    if (company != null) {
+      saleOrder.setDuration(company.getSaleConfig().getDefaultValidityDuration());
+      saleOrder.setCurrency(company.getCurrency());
```

Comment:
* sale config should be fetched using `SaleConfigService`
* the map should be filled here, not the sale order I think